at_client/lib/src/service/notification_service_impl.dart

```diff
@@ -409,4 +415,42 @@ class NotificationServiceImpl
   getStreamListenersCount() {
     return _streamListeners.length;
   }
+
+  @override
+  Future<AtNotification> fetch(String notificationId) async {
+    var notifyFetchVerbBuilder = NotifyFetchVerbBuilder()
+      ..notificationId = notificationId;
+    String? atNotificationStr;
+    try {
+      atNotificationStr = await _atClient
```

Comment:
did you test this logic and how exception is handled when there is no connectivity on client side?

Reply:
Since all the verbs thats needs to run on remote secondary needs network connectivity, have added a check - that checks for internet connectivity, in executeVerb method in RemoteSecondary.dart file. 

@VJag and @gkc : Can you plz review the change and let me know if this approach is fine?

Reply:
No. Checking for internet connectivity does not mean that the connection is still there by the time you execute the verb, the check is therefore pointless. 

I would echo what Murali asked which is, did you test the logic and how the exception is handled **when** there is no connectivity?

Reply:
Removed the condition that check for internet connectivity in remoteSecondary-> executeVerb method.
Further, tested the scenario, where the network connection is down on client side for notify fetch and `AtTimeOutException` is thrown.

```
SEVERE|2022-10-17 12:12:23.332957|AtLookup|Exception in sending to server, Exception: Waited for 10000 millis. No response after 2022-10-17 12:12:13.315518  
```

Reply:
@kalluriramkumar Have you done the follow-up you mentioned in your comment above?

Reply:
@gkc: Debugged on the issue mentioned above:
1. Observe the broken pipe issue during the data transfer when the connection is closed (shutdown secondary server). We are replicating the issue by placing the breakpoints.
2. In other scenarios getting either the successful message or the notification expiry message.

Reply:
@gkc: Debugged the issue mentioned above:
1. The broken pipe issue occurs when the connection is closed during the data transfer (Shutdown of the secondary server).
We tested this by placing the breakpoints.
2. The other cases, either the successful message or the notification expiry message is received.

Tested scenario -1 with wavi app and the following are observations:
1. The app is stable and no sign's of an app crash
2. The app logs have an error message stating the connection is closed.